### PR TITLE
MAYA-124909 - USD needs to support untextured mode like Maya does.

### PR DESCRIPTION
**Pull request by vlasovi**

### CODE REVIEW

File: lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp [hunk 1]
```diff
@@ -895,6 +899,14 @@ void HdVP2Mesh::Sync(
             _meshSharedData->_allRequiredPrimvars.push_back(HdTokens->points);
 
         _UpdatePrimvarSources(delegate, *dirtyBits, _meshSharedData->_allRequiredPrimvars);
+
+        // update the type of vertex layout to use (shared/unshared)
+        bool requireUnsharedVertexLayout
+            = _IsUnsharedVertexLayoutRequired(_meshSharedData->_primvarInfo);
+        if (_meshSharedData->_isVertexLayoutUnshared != requireUnsharedVertexLayout) {
+            _meshSharedData->_isVertexLayoutUnshared = requireUnsharedVertexLayout;
+            _ResetRenderingTopology();
```
Comment (vlasovi) [inline lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp:908]: A change in primvars may change our choice of vertex layout (shared/unshared) and then we would need to regenerate the rendering topology

**Review by JGamache-autodesk — APPROVED** [review]

File: lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp [hunk 2]
```diff
@@ -32,6 +32,7 @@
 #include <pxr/imaging/hd/basisCurves.h>
 #include <pxr/imaging/hd/changeTracker.h>
 #include <pxr/imaging/hd/enums.h>
+#include <pxr/imaging/hd/material.h>
```
[the remark this answers is not in the document]
Comment (vlasovi) [reply lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp:35]: @JGamache-autodesk ProxyRenderDelegate includes /hd/material.h to have access to HdMaterial::DirtyBits::DirtyParams enum value. HdMaterial, of course, has no knowledge of ProxyRenderDelegate, so I don't see a circular dependency here.

[the remark this answers is not in the document]
Comment (JGamache-autodesk) [reply lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp:35]: OK. My bad. Good to go then!

**Review by JGamache-autodesk — APPROVED** [review]

**Review by JGamache-autodesk — APPROVED** [review]

**Comment by vlasovi** [comment]
The failed autotest "testUsdExportPackage" [Error: Failed to delete 'w:/build/RelWithDebInfo/test/Temporary/testUsdExportPackage/tmpl8voodt1/tmp-674463F1-4382-9C03-A1D9-D1BDDF837C3A.usdc': Permission denied] is known for random failures. The PR can be merged

**Review by JGamache-autodesk — APPROVED** [review]

**Pull request merged**